lib/galaxy/tool_shed/galaxy_install/tools/tool_panel_manager.py

```diff
@@ -56,7 +56,21 @@ def add_to_shed_tool_config(self, shed_tool_conf_dict, elem_list):
                 config_elems.append(elem)
             # Add the new elements to the in-memory list of config_elems.
             for elem_entry in elem_list:
-                config_elems.append(elem_entry)
+                if elem_entry.tag == 'section':
+                    # Loop through section entries in the in-memory tool panel.
+                    for existing_elem in config_elems:
+                        # Compare the section ID for each one to the section ID for the tool being installed.
+                        if existing_elem.tag == 'section' and existing_elem.attrib.get('id', None) == elem_entry.attrib.get('id', None):
+                            # We've found a section, add the incoming tools to it.
+                            for child in elem_entry:
+                                existing_elem.append(child)
+                            # Break out of the config_elems loop back to the elem_list loop.
+                            break
+                    # If we reach this point, no section was found. Create one with contents.
+                    config_elems.append(elem_entry)
```

Comment:
```suggestion
                    else:
                        config_elems.append(elem_entry)
```
This would avoid the empty section getting added.